### ISSUE TITLE
module extension not re-evaluated when env variable is modified

### ISSUE THREAD

**Issue by laurentlb** [bazelbuild/bazel#22404]
### Description of the bug:

TL;DR: Extension modules that use ctx.getenv are not re-evaluated when the env variable is modified.

Repro.

I have a simple MODULE file:
```sh
$ cat ../repro/MODULE.bazel
my_repo = use_extension("//:ext.bzl", "my_repo")
use_repo(my_repo, "repo")
```

and the extension ext.bzl:
```starlark
def _impl(ctx):
  print("MYVAR", ctx.getenv("MYVAR"))
  # ...

my_repo = module_extension(implementation = _impl)
```

Steps:
1. During the build, notice that the print statement is correctly evaluated.
2. Modify the environment variable (e.g. `MYVAR=abc bazel build ...`)
3. Notice that the print statement is not re-executed.

I've noticed that repository rules are re-evaluated if I use `ctx.getenv`, but module extensions are not. Is it working as intended?

To me, this seems error-prone and can lead to correctness issues. I fail to see how `ctx.getenv` can be used in a safe way.

### Which category does this issue belong to?

External Dependency

### Which operating system are you running Bazel on?

MacOS

### What is the output of `bazel info release`?

release 7.1.0

**Comment by fmeum** [comment]
Can confirm that this was missed when adding `getenv`. This is still an issue with Bazel 7.2.0rc1 and at HEAD.

The Skyframe dependency is added correctly, but the information is not added to the lockfile.

CC @Wyverald
